Update ConfigurationController.java

samuelask opened the pull request:
Made minecraft not crash itself if it cant delete a file, it simply continues as normal.

Made a safer delete function (tries multiple times)

Made a safer modify function (tries multiple times)

Commit:
Update ConfigurationController.java

Made minecraft not crash itself if it cant delete a file, it simply continues as normal.

Made a safer delete function (tries multiple times)

Made a safer modify function (tries multiple times)

Commit:
Better deletion logic

Made minecraft not crash itself if it cant delete a file, it simply continues as normal.

Displays a warning if a mod cannot be deleted before minecraft launches

Made a safer delete function (tries multiple times)

Made a safer modify function (tries multiple times)

samuelask commented:
Now displays a error message before launching

![a056afb0edc4d1303e1a4aeb6500918a](https://github.com/user-attachments/assets/e2089b6d-70bc-4e9d-ba8c-975caba1bbba)

ACGaming commented:
Looks alright at first glance, I assume you've tested this sufficiently? Tell me when it's good to merge.

samuelask commented:
Yeah iv tested it, Altho it only displays the warning if it is trying to delete a file, not when it’s tries to disable it.

I can also add that functionality later but other than that it’s good to go

ACGaming commented:
If we have a mechanism like this in place, it should display a warning if any file modification should fail, so that would be appreciated.

Commit:
Improved error feedback

- Added warning display to all modify functions

- Cleaned up and made the warning display more readable for long errors

samuelask commented:
I finished it, added warning for every failed file modification.
Iv also tested it again sufficiently

Commit:
logging fix

- ReAdded logging of Moving/Renaming file

ACGaming commented:
Obliged!
